atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1814,6 +1828,8 @@ private void commitWrites(TransactionService transactionService) {
                         "commitPutCommitTs",
                         () -> putCommitTimestamp(commitTimestamp, commitLocksToken, transactionService));
 
+                lockWatchValueCache.updateCacheOnCommit(cache.getDigest(), getStartTimestamp());
+
```

Comment:
This might live better in the same place that `LockWatchEventCache` does its update on commit, because then we can batch (this is the central cache you're updating, which means this could slow things done if we're not careful).

Reply:
You mean in the CommitTimestampGetters? That probably makes sense, yeah. Just to double check -- we only actually update with the reads, so even if the transaction fails after we do this update, it's still correct, right?

Reply:
the former: yes (I think that's what it is called these days).

the latter: only update reads, and that _should_ happen after we verify the PCC, so if there were R-W conflicts, those reads should not end up in the cache either (we'll have to be careful about which part of the lifecycle that happens in).

It is worth noting that, internally, we actually only update the cache once the transaction has fully committed successfully, but I think it is technically fine to do it during commit here (with the caveat above that the PCC is done at the correct time).